### PR TITLE
Doc Fix: Passing `datamodule` argument to `trainer.tune` is supported

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/8753

### Does your PR introduce any breaking changes? If yes, please list them.

No

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7 @akihironitta

### CODE REVIEW

**Comment by krshrimali** [comment]
Not happy with this change. Working on it. The example mentioned in the doc currently doesn't work. Maybe it's not complete, and in that case, we should mention that other required methods need to be implemented.

**Review by akihironitta — APPROVED** [review]
LGTM 🚀

**Comment by akihironitta** [comment]
>  The example mentioned in the doc currently doesn't work.

@krshrimali Could be related to #11604?

**Comment by krshrimali** [comment]
> > The example mentioned in the doc currently doesn't work.
> 
> @krshrimali Could be related to #11604?

@akihironitta - Yes, does look related! :) Thank you for pointing to the issue.

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/advanced/training_tricks.rst [hunk 1]
```diff
@@ -251,8 +256,8 @@ with the parameters of the algorithm. This can be done by invoking the
     # update hparams of the model
     model.hparams.lr = new_lr
 
-    # Fit model
-    trainer.fit(model)
+    # Fit model, with a datamodule (assuming it's defined before)
```
[the remark this answers is not in the document]
Comment (rohitgr7) [reply docs/source/advanced/training_tricks.rst:259]: this code is just for representing how it works, and won't work anyways even without a datamodule.
I'd say instantiate datamodule too like:
```py
model = MyModelClass(hparams)
datamodule = MyDataModuleClass(hparams)
```
also, you might need to fix the complete example since it doesn't align properly.

[the remark this answers is not in the document]
Comment (krshrimali) [reply docs/source/advanced/training_tricks.rst:259]: Thanks, @rohitgr7 for the comment - don't want to define the `datamodule` here, since the data modules are discussed towards the end of the page (in the documentation). However, I agree we can fix the examples but let's keep it for another PR. Since this PR intends to give a hint that passing a data module is valid and also links to the doc for `tune` API. Let me know if you disagree.

[the remark this answers is not in the document]
Comment (krshrimali) [reply docs/source/advanced/training_tricks.rst]: Post offline discussion with @rohitgr7 - we agreed that we want to extend the note showing the usage of `datamodule`. Also, there shouldn't be a need to fix the calls for each instance of `trainer.fit(model)/trainer.tune(model)` to `trainer.fit(model, datamodule)/trainer.tune(model, datamodule)`.

The PR has been updated. Thanks, @rohitgr7. Also, thanks @kaushikb11 for the question :)

File: docs/source/advanced/training_tricks.rst [hunk 2]
```diff
@@ -189,11 +218,15 @@ initial lr.
...
-which can be accessed via ``self.learning_rate`` or ``self.lr``.
+To enable the learning rate finder, your :doc:`lightning module <../common/lightning_module>` needs to
+have a ``learning_rate`` or ``lr`` attribute (or as a field in your ``hparams`` i.e.
+``hparams.learning_rate`` or ``hparams.lr``). Then, set ``Trainer(auto_lr_find=True)``
+during trainer construction, and then call ``trainer.tune(model)`` to run the LR finder.
+The suggested ``learning_rate`` will be written to the console and will be automatically
+set to your :doc:`lightning module <../common/lightning_module>`, which can be accessed
+via ``self.learning_rate`` or ``self.lr``.
+
+.. seealso:: :ref:`trainer.tune <common/trainer:tune>`.
```
Comment (rohitgr7) [inline docs/source/advanced/training_tricks.rst:229]: I hope this is rendered correctly. Can you check, once the doc build is fixed?

**Review by rohitgr7 — APPROVED** [review]
noice!!

**Review by carmocca — APPROVED** [review]
Thanks!

**Review by akihironitta — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
